agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitMultiProjectPropertiesLocator.java

```diff
@@ -41,16 +42,16 @@ public GitMultiProjectPropertiesLocator(DelayedTeamscaleMultiProjectUploader upl
 	 * Asynchronously searches the given jar file for a git.properties file.
 	 */
 	@Override
-	public void searchJarFileForGitPropertiesAsync(File jarFile) {
-		executor.execute(() -> searchJarFile(jarFile));
+	public void searchJarFileForGitPropertiesAsync(Pair<File, Boolean> file) {
+		executor.execute(() -> searchJarFile(file));
 	}
 
-	private void searchJarFile(File jarFile) {
-		logger.debug("Searching jar file {} for multiple git.properties", jarFile.toString());
+	private void searchJarFile(Pair<File, Boolean> file) {
+		logger.debug("Searching jar file {} for multiple git.properties", file.toString());
 		try {
-			ProjectRevision data = GitPropertiesLocatorUtils.getProjectRevisionFromGitProperties(jarFile);
+			ProjectRevision data = GitPropertiesLocatorUtils.getProjectRevisionFromGitProperties(file.getFirst(), file.getSecond());
 			if (data == null) {
-				logger.debug("No git.properties file found in {}", jarFile.toString());
+				logger.debug("No git.properties file found in {}", file.toString());
```

Comment:
Shouldn't that become

```suggestion
				logger.debug("No git.properties file found in {}", file.getFirst().toString());
```

I would also suggest to rename file to something that better describes what the pair represents